net-mgmt/pfSense-pkg-lldpd/files/usr/local/pkg/lldpd/lldpd.inc

```diff
@@ -0,0 +1,184 @@
+<?php
+/*
+ * lldpd.inc
+ *
+ * part of pfSense (https://www.pfsense.org)
+ * Copyright (c) 2018 Denny Page
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+require_once("config.inc");
+require_once("functions.inc");
+require_once("util.inc");
+require_once("service-utils.inc");
+
+define("LLDPD_RCFILE", "/usr/local/etc/rc.d/lldpd.sh");
+define("LLDPD_DAEMON", "/usr/local/sbin/lldpd");
+define("LLDPD_CLIENT", "/usr/local/sbin/lldpcli");
+
+$shortcut_section = "lldpd";
+
+
+function lldpd_sync_config() {
+	global $config;
+
+	if (is_service_running('lldpd')) {
+		log_error("Stopping service lldpd");
+		stop_service('lldpd');
+	}
+
+	$lldpd_config = &$config['installedpackages']['lldpd']['config'][0];
+	if (isset($lldpd_config['enable'])) {
+		$enable = $lldpd_config['enable'];
+	} else {
+		$enable = 'no';
+	}
+
+	if ($enable !== 'yes') {
+		unlink_if_exists(LLDPD_RCFILE);
+		return;
+	}
+
+	$cmd = LLDPD_DAEMON;
+	
+	/* LLDP protocol */
+	switch ($lldpd_config['lldp_proto']) {
+		case 'active':
+			$cmd .= ' -l';
+			break;
+		case 'disabled':
+			$cmd .= ' -ll';
+			break;
+	}
```

Comment:
All switch statements must have a default case, even if it's a no-op, to avoid ambiguity.